gcloud/monitoring/client.py

```diff
@@ -195,6 +203,121 @@ def metric_descriptor(self, type_,
             display_name=display_name,
         )
 
+    @staticmethod
```

Comment:
Those are some fair points as to some other good use cases of the decorator

Either way, I believe in the interest of consistency these methods belong in the class. If we move them to public methods, might as well just kill them all together, since the entire point was to create a consistent API for creating API objects from the client.  @rimey can chime in if he has an opinion.